docs/config/markdown-math-filter.sed

```diff
@@ -0,0 +1,25 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Doxygen filter to format markdown math for Doxygen
+#
+# In Doxygen documentation, including markdown files or Jupyter notebooks that
+# get converted to markdown, you can use standard $...$ syntax for inline math,
+# and the following syntax for display-style math:
+#
+# $$\begin{equation}
+# ...
+# \end{equation}$$
+#
+# You can also replace `equation` with another supported environment, e.g.
+# `align`.
+#
+# This Doxygen filter formats the math for Doxygen, i.e., uses \f$...\f$ for
+# inline math and \f{equation}{ ... \f} for display-style math.
+
+# Wrap display math in \f{equation}{ ... \f}
+s|\$\$\\begin{\([^}]*\)}|\\f{\1}{|g
+s|\\end{\([^}]*\)}\$\$|\\f}|g
+
+# Wrap inline math in \f$...\f$
+s|\([^\\f]\)\$\([^$]*\)\$|\1\\f\$\2\\f\$|g
```

Comment:
What is excluding `\$` from the match for?  That doesn't appear anywhere in our markdown source.

Reply:
That makes the regex "lazy", so it won't gobble up `$` characters until it hits the last one on the line.

Reply:
Sorry, I don't follow.

Reply:
switched to perl, so the syntax here changed (all of this is way easier in perl)

Reply:
Unfortunately, perl scripts are banned from SpECTRE.  Because python is better or some such nonsense.

Reply:
Hm I'm not sure Python can be as fast as perl here, haven't tried though..

Reply:
Full docs generation takes 102s with no filter, 108s with Perl filter, 135s with Python filter. Thoughts?

Reply:
I'm not surprised python is slow.  The interpreter takes tens of milliseconds to start, about an order of magnitude slower than perl.

The anti-perl faction is over there -> @nilsdeppe @kidder

Reply:
The reasoning isn't "Python is better than Perl" (this is super subjective and application dependent!) the reasoning is that more science folks (or at least students) are comfortable with Python than perl (python is actually taught, while perl isn't), as well as that having fewer languages being used in the code base makes it more accessible. 

Having said all that, we have used single-line perl regex calls because that's faster and easier than Python. I haven't looked at how long this script would be, but I'm okay with fairly short and simple perl files if the reason is performance (33s overhead on dox build is too much IMO). What I would like to avoid is the situation in SpEC where we have something like 50% of scripts in python and 50% in perl, then have them calling back and forth. This results in very difficult to understand code, regardless of what languages are being used. It looks like this sed script could just be a few one-line perl regex calls from CMake. We used to do a lot of regex-replace in perl for Charm things, if I remember correctly.

Reply:
👍 I second all that